EIPS/eip-00.md

```diff
@@ -0,0 +1,789 @@
+
+* Inability to use `tx.origin` or contracts that rely on it as it returns the meaningless address of a bundler.
+
+EIP-2938 defines a very mature alternative approach to Account Abstraction. However, it does not translate
+well to the architecture of ERC-4337 that is being used in production without any protocol changes.
+Therefore, the implementation of EIP-2938 will not benefit as much from the production experience gained by using
+ERC-4337 and from maintaining backward compatibility with it.
+
+There is also a possibility that at some point in the future, the EOAs on Ethereum will be replaced with pre-deployed
+smart contracts. This, however, is impossible without an addition of Native Account Abstraction to the protocol.
+
+## Specification
+
+### Constants
+
+| Name                  | Value                                                                           |
+|-----------------------|---------------------------------------------------------------------------------|
+| FORK_BLOCK            | 	TBD                                                                            |
+| AA_TX_TYPE            | 4                                                                               |
+| AA_ENTRY_POINT        | `address(9999)` << current EIP number                                           |
+| AA_SENDER_CREATOR     | `address(ffff9999)`                                                             |
+| AA_NONCE_MANAGER      | TODO                                                                            |
+| AA_BASE_GAS_COST      | 15000                                                                           |
+| AA_ECRECOVER_COST     | 6000                                                                            |
+| VERSION               | 1                                                                               |
+| MAGIC_VALUE_SENDER    | 0xbf45c166  // keccak256("validateTransaction(uint256,bytes32,bytes)")          |
+| MAGIC_VALUE_PAYMASTER | 0xe0e6183a  // keccak256("validatePaymasterTransaction(uint256,bytes32,bytes)") |
+| MAX_CONTEXT_SIZE      | 65536                                                                           |
+
+### New Transaction Type
+
+A new [EIP-2718](./eip-2718) transaction with type AA_TX_TYPE is introduced. Transactions of this type are referred to as
+“AA transactions”. Their payload should be interpreted as:
+
+```
+
+0x04 || 0x00 || rlp([
+  chainId, sender, nonce, builderFee,
+  callData, paymasterData, deployerData,
+  maxPriorityFeePerGas, maxFeePerGas,
+  validationGasLimit, paymasterGasLimit, callGasLimit,
+  accessList, signature
+])
+
+```
+
+The base gas cost of this transaction is set to AA_BASE_GAS_COST instead of 21000 to reflect the lack of “intrinsic”
+ECDSA and signature.
+
+If `paymasterData` is specified, its first 20 bytes contain the address of a `paymaster` contract.
+
+If `deployerData` is specified, its first 20 bytes contain the address of a `deployer` contract.
```

Comment:
why not have `deployerAddr` and `deployerData` (same for paymaster) instead?

Reply:
> why not have `deployerAddr` and `deployerData` (same for paymaster) instead?

One reason we combined the address and the data passed to it is that this is what the ERC-4337 does.
"Packing" these four fields into two saved some calldata for the bundlers.
This is not as relevant for the native `TransactionType4`.
However, many transactions don't have a `Paymaster`, and almost all of the transactions don't have a `deployerData` as only the first transaction by an address has it. 
This optimization remains a somewhat valid concern for the native `TransactionType4` as loading the `TransactionType4` into memory and passing it to a call will be a little more expensive.
If 1% of the transactions have a `Deployer`, there is no reason to waste data on another address.